packages/react/src/components/Header/Header.tsx

```diff
@@ -18,6 +18,10 @@ import { Accessibility } from '../../lib/accessibility/types'
 import { defaultBehavior } from '../../lib/accessibility'
 import { ReactProps, ShorthandValue } from '../../types'
 
+export interface HeaderSlotClassNames {
+  description: string
```

Comment:
a very important thing is the contract that stays behind these class names: what we are promising to the client is that in the rendered tree there will be always a `& .${slotClassNames.description}` element rendered in case if `description` is provided.

However, the following example will break these guarantees:
```jsx
<Header description={<Text>Some description</Text>} ... />
```
![image](https://user-images.githubusercontent.com/9024564/54932504-34cbe600-4f1b-11e9-95d4-2541ef84e62a.png)


As we might see, there is no `.ui-header__description` element rendered in the tree, while `description` was actually provided. This problem would occur anytime one would use different component type for the slot.

See this to be an important thing to fix before merge.

Reply:
This is definitely a problem but it's not introduced by this API. We already have this behavior for all the cases where we're using the `slotClassNames` approach. I cannot think of a solution for this rather than telling clients to avoid children API?

@kuzhelov @mnajdova

Reply:
discussed with @kuzhelov will fix